src/Microsoft.AspNetCore.DataProtection/TypeForwardingActivator.cs

```diff
@@ -39,16 +40,14 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
                 forwardedTypeName = originalTypeName.Replace(OldNamespace, CurrentNamespace);
             }
 
-#if NET46
-            if (candidate || forwardedTypeName.Contains(CurrentNamespace))
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Core"))
```

Comment:
When we added this code in #223, there goal was to make this fix as conservative as possible. There is no need to strip version numbers on .NET Core. 

But now, given that we have yet-to-determine a reliable way to detect .NET Core vs .NET Framework w/o cross-compiling, IMO we are better off always stripping the version number.